app/assets/javascripts/discourse/app/components/sidebar/common/community-section.js

```diff
@@ -13,45 +13,68 @@ export default class SidebarCommunitySection extends Component {
   @service appEvents;
   @service siteSettings;
 
-  // Override in child
-  defaultMainSectionLinks = [];
-  defaultAdminMainSectionLinks = [];
-  defaultMoreSectionLinks = [];
-  defaultMoreSecondarySectionLinks = [];
   headerActionsIcon;
   headerActions;
+  sectionLinks;
+  moreSectionLinks;
+  moreSecondarySectionLinks;
+  callbackId;
 
-  get moreSectionLinks() {
-    return [...this.defaultMoreSectionLinks, ...customSectionLinks].map(
-      (sectionLinkClass) => {
-        return this.#initializeSectionLink(sectionLinkClass);
-      }
-    );
-  }
+  constructor() {
+    super(...arguments);
+
+    this.moreSectionLinks = [
+      ...this.defaultMoreSectionLinks,
+      ...customSectionLinks,
+    ].map((sectionLinkClass) => {
+      return this.#initializeSectionLink(sectionLinkClass);
+    });
 
-  get moreSecondarySectionLinks() {
-    return [
+    this.moreSecondarySectionLinks = [
       ...this.defaultMoreSecondarySectionLinks,
       ...secondaryCustomSectionLinks,
     ].map((sectionLinkClass) => {
       return this.#initializeSectionLink(sectionLinkClass);
     });
-  }
 
-  get mainSectionLinks() {
-    return this.currentUser?.staff
+    const mainSectionLinks = this.currentUser?.staff
       ? [...this.defaultMainSectionLinks, ...this.defaultAdminMainSectionLinks]
       : [...this.defaultMainSectionLinks];
-  }
 
-  get sectionLinks() {
-    return this.mainSectionLinks.map((sectionLinkClass) => {
+    this.sectionLinks = mainSectionLinks.map((sectionLinkClass) => {
       return this.#initializeSectionLink(sectionLinkClass);
     });
+
+    this.callbackId = this.topicTrackingState.onStateChange(() => {
+      this.sectionLinks.forEach((sectionLink) => {
```

Comment:
The only thing I would worry about here is if a `sectionLink` defines particularly slow `onTopicTrackingStateChange` code. For example, in the teams sidebar I had to end up writing a `debounce` into this callback because it was causing performance issues:

```javascript
watchTopicTrackingState() {
  this.topicTrackingState.onStateChange(this.stateUpdated.bind(this));
},

stateUpdated() {
  // going through topicTrackingState.forEachTracked and updating all
  // the counts is quite an expensive operation, so we do not want to
  // do it excessively
  discourseDebounce(this, this._updateSidebarCounts, 100);
},
```

Reply:
O I resolved the problem in https://github.com/discourse/discourse/commit/30bd1dcefd1a175c587bce5ba063869a0e696b07 previously so the debounce is no longer necessary.

Reply:
Ah very nice @tgxworld thanks for that!